docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
-    <RelatedReadItem path="/develop/typescript/message-passing#signals" text="Send and handle Signals with the TypeScript SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/dotnet/message-passing#define-a-signal" text="Send and handle Signals with the .NET SDK" archetype="feature-guide" />
-</RelatedReadContainer>
+From the above descriptions, it may sound like Updates can do all the things that Queries and Signals can do. But, unlike Signals, Updates must be synchronous, that is, they must wait for the worker running the Workflow to acknowledge the request.
+
+Use Signals instead of Updates when:
+
+- You want to send a fully asynchronous message.
+- Your client does not want to rely on the worker being available.
+- Your Workflow’s client cares more about the latency of sending the message than it does of the end-to-end operation.
+- When you need to [Signal-With-Start](signal-with-start) atomically start the Workflow and then Signal as the first operation. (since there is currently no Update-with-start operation.)
+
+Use Updates instead of Signals when:
+
+- You want to track the completion of the message from the client.
+- You need a result or an exception from your message without having to query subsequently.
+- You’d like to “validate” the Update before accepting it into the Workflow and its history.
+- Your client cares more about the latency of the end-to-end operation than you care about the latency of sending the message.
+
+### For read requests
 
-A Signal delivers data to a running Workflow Execution.
-It cannot return data to the caller; to do so, use a [Query](#queries) instead.
-The Workflow code that handles a Signal can mutate Workflow state.
-A Signal can be sent from a Temporal Client or a Workflow.
-When a Signal is sent, it is received by the Temporal Service and recorded as an Event to the Workflow Execution [Event History](/workflows#event-history).
-A successful response from the Temporal Service means that the Signal has been persisted and will be delivered at least once to the Workflow Execution.[^1]
-The next scheduled Workflow Task will contain the Signal Event.
+You normally want to do a Query:
 
-A Signal must include a destination (Namespace and Workflow Id) and name.
-It can include a list of arguments.
+- Queries are efficient–they never add entries to the [Workflow Event History](/workflows#event-history).
+- Queries can operate on completed Workflows.
 
-Signal handlers are Workflow functions that listen for Signals by the Signal name.
-Signals are delivered in the order they are received by the Temporal Service and written to History.
-If multiple deliveries of a Signal would be a problem for your Workflow, add idempotency logic to your Signal handler that checks for duplicates.
+However, because Queries cannot block, sometimes Updates are best.\
+When your goal is to do a read once the Workflow achieves a certain desired state, you have two options.
```

Comment:
```suggestion
When your goal is to do a read once the Workflow achieves a certain desired state, you have two options:
```

Reply:
There are more than two options: another one that we currently document is that the client-side application could run an activity worker to receive responses from signals: https://github.com/temporalio/samples-go/tree/main/reqrespactivity

Reply:
Interesting, I'm going to pass on this suggestion as I've not heard of anybody doing this and it feels a bit off topic, but feel free to add it as part of your edits if you think I'm off here.

Reply:
My suggestion (which I didn't make clear) is probably just to say "you have at least two options" or "options include" or something.

Reply:
(That pattern is interesting though.)